app/controllers/shared/api.php

```diff
@@ -290,7 +290,7 @@
             $minimumFactors = ($mfaEnabled && $hasMoreFactors) ? 2 : 1;
 
             if (!in_array('mfa', $route->getGroups())) {
-                if ($session && \count($session->getAttribute('factors')) < $minimumFactors) {
+                if ($session && !empty($session->getAttribute('factors')) && \count($session->getAttribute('factors')) < $minimumFactors) {
```

Comment:
Might be simpler to pass an empty array as the default for getAttribute inside count:

```suggestion
                if ($session && \count($session->getAttribute('factors', [])) < $minimumFactors) {
```